File: app/src/main/java/com/example/android/rtdp/implementation/java/data/NetInfoRepository.java
package com.example.android.rtdp.implementation.java.data;

import android.app.Application;

import androidx.lifecycle.LiveData;
import androidx.lifecycle.MutableLiveData;

import com.example.android.rtdp.implementation.java.data.model.NetInfo;
import com.example.android.rtdp.implementation.java.data.service.MainService;
import com.example.android.rtdp.implementation.java.data.source.remote.RemoteDataSource;
import com.example.android.rtdp.implementation.java.data.source.remote.RemoteResponseHandler;
import com.example.android.rtdp.implementation.java.data.source.remote.Response;

import java.util.Random;

import javax.inject.Inject;

public final class NetInfoRepository extends BaseRepository {
    private static final int BOUND = 30;
    private static final Random RANDOM = new Random();

    private final MainService mService;

    @Inject
    public NetInfoRepository(Application app, MainService service) {
        super(app);
        mService = service;
    }

    public LiveData<Response<NetInfo>> getNetInfo() {
        int id = RANDOM.nextInt(BOUND) + 1;
        MutableLiveData<Response<NetInfo>> liveData = new MutableLiveData<>();
        RemoteDataSource<NetInfo> remoteSource = new RemoteDataSource<>(mService.getNetInfo(id));
        remoteSource.execute(new RemoteResponseHandler<>(getApp(), liveData));
        addRemoteDataSource(remoteSource);
        return liveData;
    }
}

File: app/src/main/java/com/example/android/rtdp/implementation/kotlin/data/NetInfoRepository.kt
package com.example.android.rtdp.implementation.kotlin.data

import androidx.lifecycle.LiveData
import com.example.android.rtdp.implementation.kotlin.data.model.NetInfo
import com.example.android.rtdp.implementation.kotlin.data.service.MainService
import com.example.android.rtdp.implementation.kotlin.data.source.remote.Response
import java.util.*
import javax.inject.Inject

class NetInfoRepository @Inject constructor(private val service: MainService) {
    fun getNetInfo(): LiveData<Response<NetInfo>> =
            service.getNetInfo(id = Random().nextInt(BOUND) + 1)
}

private const val BOUND = 30